prow/pjutil/trigger.go

```diff
@@ -0,0 +1,125 @@
+	} else if len(prowJob.Spec.ExtraRefs) > 0 {
+		identifier = fmt.Sprintf("%s/%s", prowJob.Spec.ExtraRefs[0].Org, prowJob.Spec.ExtraRefs[0].Repo)
+	} else {
+		return "failed to extract decoration config identifier"
+	}
+	spec := downwardapi.NewJobSpec(prowJob.Spec, prowJob.Status.BuildID, prowJob.Name)
+	jobBasePath, _, _ := gcsupload.PathsForJob(config.Plank.GetDefaultDecorationConfigs(identifier).GCSConfiguration, &spec, "")
+	return fmt.Sprintf("%s%s/%s",
+		config.Deck.Spyglass.GCSBrowserPrefix,
+		config.Plank.GetDefaultDecorationConfigs(identifier).GCSConfiguration.Bucket,
+		jobBasePath,
+	)
+}
+
+type prowjobResult struct {
+	Status       pjapi.ProwJobState `json:"status"`
+	ArtifactsURL string             `json:"prowjob_artifacts_url"`
+	URL          string             `json:"prowjob_url"`
+}
+
+func resultForJob(pjclient prowv1.ProwJobInterface, selector string) (*prowjobResult, *pjapi.ProwJob, bool, error) {
+	w, err := pjclient.Watch(context.TODO(), metav1.ListOptions{FieldSelector: selector})
+	if err != nil {
+		return nil, nil, false, fmt.Errorf("failed to create watch for ProwJobs: %w", err)
+	}
+
+	for event := range w.ResultChan() {
+		prowJob, ok := event.Object.(*pjapi.ProwJob)
+		if !ok {
+			return nil, nil, false, fmt.Errorf("received an unexpected object from Watch: object-type %s", fmt.Sprintf("%T", event.Object))
+		}
+
+		switch prowJob.Status.State {
+		case pjapi.FailureState, pjapi.AbortedState, pjapi.ErrorState, pjapi.SuccessState:
+			return &prowjobResult{
+				Status: prowJob.Status.State,
+				URL:    prowJob.Status.URL,
+			}, prowJob, false, nil
+		}
+	}
+	return nil, nil, true, nil
+}
+
+// TriggerProwJob would trigger the job provided by the prowjob parameter
+func TriggerProwJob(o prowflagutil.KubernetesOptions, prowjob *pjapi.ProwJob, config *prowconfig.Config, envVars map[string]string, dryRun bool) error {
+	logrus.Info("getting cluster config")
+	pjclient, err := o.ProwJobClient(config.ProwJobNamespace, dryRun)
+	if err != nil {
+		return fmt.Errorf("failed getting prowjob client: %w", err)
+	}
+
+	logrus.WithFields(ProwJobFields(prowjob)).Info("submitting a new prowjob")
+	created, err := pjclient.Create(context.TODO(), prowjob, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to submit the prowjob: %w", err)
+	}
+
+	logger := logrus.WithFields(ProwJobFields(created))
+	logger.Info("submitted the prowjob, waiting for its result")
+
+	selector := fields.SelectorFromSet(map[string]string{"metadata.name": created.Name})
+
+	var result *prowjobResult
+	var shouldContinue bool
+	var prowJob *pjapi.ProwJob
+	for {
+		result, prowJob, shouldContinue, err = resultForJob(pjclient, selector.String())
+		if err != nil {
+			return fmt.Errorf("failed to watch job: %w", err)
+		}
+		if !shouldContinue {
+			break
+		}
+	}
+	if result.Status != pjapi.SuccessState {
+		logrus.Warn("job failed")
+	}
+	result.ArtifactsURL = getJobArtifactsURL(prowJob, config)
```

Comment:
Can we only use the spyglass url in `.satus.url` as steve suggested and save ourselves the calculation of the artifacts url? There is a link to the latter in the former

Reply:
I dont know why we were calculating it, but I left it in since it was because it was explicitly calculated in the cvp-trigger tool. I am removing it for now but we might need to revisit this later if it is necessary in cvp-trigger